packages/launcher/src/AppUpdater.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { Typography, Box, CircularProgress } from '@material-ui/core';
+import { ipcRenderer } from 'electron';
+import App from '@qilin/shared/src/App';
+import { BACKGROUND_DARK } from '@qilin/shared/src/styles/colors';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import { useTranslation } from 'react-i18next';
+
+import {
+  CHECK_FOR_UPDATE_PENDING,
+  CHECK_FOR_UPDATE_SUCCESS,
+  CHECK_FOR_UPDATE_FAILURE,
+  DOWNLOAD_UPDATE_FAILURE,
+  DOWNLOAD_UPDATE_PENDING,
+  DOWNLOAD_UPDATE_SUCCESS,
+  QUIT_AND_INSTALL_UPDATE,
+  APP_INFO,
+} from './ipc.constants';
+
+const useStyle = makeStyles({
+  root: {
+    backgroundColor: BACKGROUND_DARK,
+    height: '100vh',
+    color: '#fff',
+    flexDirection: 'column',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  appInfoContainer: {
+    padding: 10,
+    position: 'absolute',
+    right: 0,
+    top: 0,
+  },
+  loader: {
+    marginTop: 15,
+    color: '#fff',
+  },
+});
+
+const AppUpdater = () => {
+  const { t } = useTranslation();
+  const [loading, setLoading] = useState(true);
+  const [readyToDownload, setReadyToDownload] = useState(false);
+  const [info, setAppInfo] = useState<{ name: string; version: string; channel: string; channels: [] } | null>(null);
+  const [status, setUpdateStatus] = useState(t('update_status.checking'));
+  const [updateError, setUpdateError] = useState<{ code: string; description: string } | null>(null);
+  const [redirectToApp, setRedirectToApp] = useState(false);
+
+  const classes = useStyle();
+
+  const downloadUpdateAndInstall = () => {
+    setUpdateStatus(t('update_status.downloading_and_install'));
+    ipcRenderer.send(DOWNLOAD_UPDATE_PENDING);
+  };
+
+  useEffect(() => {
+    ipcRenderer.send(APP_INFO);
+
+    ipcRenderer.on(APP_INFO, (event: any, appInfo) => {
```

Comment:
Мне кажется обработчики надо сохранить в константы и вынести из useEffect, типа
```suggestion
    ipcRenderer.on(APP_INFO, handleAppInfo)
```

Reply:
работать будет одинаково)
но можно вынести